### PR TITLE
Remove the appended domain name from roles in the ID token

### PR DESCRIPTION
**Pull request by ChanikaRuchini**
### Proposed changes in this pull request
$subject
Fixes - [#11189](https://github.com/wso2/product-is/issues/11189)

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java [hunk 1]
```diff
@@ -450,6 +454,25 @@ private void handleEmailVerifiedClaim(Map<String, Object> returnClaims) {
         }
     }
 
+    private void handleRolesClaim(Map<String, Object> returnClaims) {
+
+        if (returnClaims.containsKey(ROLES) && IdentityUtil.isGroupsVsRolesSeparationImprovementsEnabled()) {
+            String delim = "";
+            String separator = ",";
```
Comment (ashensw) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java:461]: Use `MultiAttributeSeparator` here.

Comment (ashensw) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java:461]: Let's use `MultiAttributeSeparator` here.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java [hunk 2]
```diff
@@ -450,6 +454,25 @@ private void handleEmailVerifiedClaim(Map<String, Object> returnClaims) {
...
+            String[] roles = returnClaims.get(ROLES).toString().split(separator);
+            StringBuffer multiValuedStringBf = new StringBuffer();
+
+            for (String role : roles) {
+                if (UserCoreConstants.INTERNAL_DOMAIN.equalsIgnoreCase(IdentityUtil.extractDomainFromName(role))) {
+                    role = UserCoreUtil.removeDomainFromName(role);
+                }
+                multiValuedStringBf.append(delim).append(role);
+                delim = separator;
+            }
```
Comment (ChamathNS) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java:471]: we can use `StringUtils.join(roles, ",")` here

Comment (ChamathNS) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java:471]: we can use Java 8 filters to get the roles without the domain

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java (hunk 1 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java]: Resolved

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java (hunk 2 above)
Comment (ChanikaRuchini) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/OpenIDConnectClaimFilterImpl.java]: Fixed

**Review by ashensw — APPROVED** [review]

**Pull request merged**